### PR TITLE
switch bv2integer and integer2bv to irep_idt

### PR DESCRIPTION
**Pull request by kroening**
- [X] Each commit message has a non-empty body, explaining why the change was made.
- [X] My contribution is formatted in line with CODING_STANDARD.md.
- [ ] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [ ] My commit message includes data points confirming performance improvements (if claimed).
- [X] My PR is restricted to a single feature or bugfix.
- [X] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: 58780eb18730560fdd166c78bca24952d44ee214).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/87110647

File: src/util/arith_tools.h [hunk 1]
```diff
@@ -164,4 +164,10 @@ mp_integer power(const mp_integer &base, const mp_integer &exponent);
 void mp_min(mp_integer &a, const mp_integer &b);
 void mp_max(mp_integer &a, const mp_integer &b);
 
+/// convert an integer to bit-vector representation with given width
```
Comment (tautschnig) [inline src/util/arith_tools.h]: While at it: can we remove the duplicate comment, going just with the one with the definition?

**Review by tautschnig — APPROVED** [review]
As also noted elsewhere: are `2bv` and `bv2` actually an adequate suffix/prefix? With #3107 it's clear that these aren't necessarily vectors of bits, but really it's about the internal representation of numeric values.

**Comment by kroening** [comment]
bvrep? bitvector_representation is long.

**Comment by tautschnig** [comment]
I think "bvrep" is a good compromise!

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: 97b6ab5cd9f6499a17b95cc09aabcc6b68dc051c).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/87128345

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: e57cdc50ea44b4f6b060e9336eb0e2806e1b23d3).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/87137261

**Review by tautschnig — APPROVED** [review]

**Review by allredj — COMMENTED** [review]
This PR failed Diffblue compatibility checks (cbmc commit: 1edaf0d70f77313e1de31acd3e657042f6dae9e0).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/87168314
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: fcd71974209cff5ec38517ed5ad4985baf369881).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/87853945

File: jbmc/src/java_bytecode/java_bytecode_concurrency_instrumentation.cpp [hunk 2]
```diff
@@ -86,7 +86,7 @@ static const std::string get_thread_block_identifier(
 {
   PRECONDITION(f_code.arguments().size() == 1);
   const exprt &expr = f_code.arguments()[0];
-  mp_integer lbl_id = bv2integer(expr.op0().get_string(ID_value), false);
+  mp_integer lbl_id = bvrep2integer(expr.op0().get_string(ID_value), false);
```
Comment (smowton) [inline jbmc/src/java_bytecode/java_bytecode_concurrency_instrumentation.cpp]: get_string -> get

**Review by smowton — COMMENTED** [review]
One missing conversion

File: jbmc/src/java_bytecode/java_bytecode_concurrency_instrumentation.cpp (hunk 2 above)
Comment (kroening) [reply jbmc/src/java_bytecode/java_bytecode_concurrency_instrumentation.cpp]: done

**Review by smowton — APPROVED** [review]

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: 39bdf533ce9a6a22cc3175f1818e7e365ac2d0c1).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/88439677

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: b2390e41ce036c2dd52fa04ee8bbd3def4a063dd).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/88528654

**Pull request merged**